src/objects/Player.gd

```diff
@@ -54,36 +71,43 @@ func _physics_process(delta: float) -> void:
 			if iframe_timer <= 0.0:
 				is_vulnerable = true
 
-		var wishdir := Vector2.ZERO
-		var jump_pressed := false
-		if should_control():
-			wishdir = Input.get_vector("move_left", "move_right", "move_backwards", "move_forwards")
-			jump_pressed = Input.is_action_just_pressed("jump")
+		if (not get_tree().network_peer) or is_network_master():
+			var wishdir := Vector2.ZERO
+			var jump_pressed := false
+			if should_control():
+				wishdir = Input.get_vector("move_left", "move_right", "move_backwards", "move_forwards")
+				jump_pressed = Input.is_action_just_pressed("jump")
 
-		var forward_vector := Vector3.FORWARD.rotated(Vector3.UP, rotation.y)
-		var right_vector := Vector3.FORWARD.rotated(Vector3.UP, rotation.y - PI / 2)
+			var forward_vector := Vector3.FORWARD.rotated(Vector3.UP, rotation.y)
+			var right_vector := Vector3.FORWARD.rotated(Vector3.UP, rotation.y - PI / 2)
 
-		var move_vector := wishdir.x * right_vector + wishdir.y * forward_vector
+			var move_vector := wishdir.x * right_vector + wishdir.y * forward_vector
 
-		velocity.x = move_vector.x * MOVE_SPEED
-		velocity.z = move_vector.z * MOVE_SPEED
+			velocity.x = move_vector.x * MOVE_SPEED
+			velocity.z = move_vector.z * MOVE_SPEED
 
-		var jumping := false
+			var jumping := false
 
-		if is_on_floor() and jump_pressed:
-			jumping = true
-			velocity.y = JUMP_POWER
+			if is_on_floor() and jump_pressed:
+				jumping = true
+				velocity.y = JUMP_POWER
 
-		velocity.y -= delta * GRAVITY
-		velocity = move_and_slide_with_snap(velocity, Vector3.ZERO if jumping else Vector3.DOWN, Vector3.UP, true)
+			velocity.y -= delta * GRAVITY
+			velocity = move_and_slide_with_snap(velocity, Vector3.ZERO if jumping else Vector3.DOWN, Vector3.UP, true)
 
 	if get_tree().network_peer and is_network_master():
-		rpc_unreliable("set_network_transform", translation, rotation)
+		rpc_unreliable("set_network_transform", translation, head.global_rotation)
+
+
+func _process(_delta: float) -> void:
+	var interp_translation := get_global_transform_interpolated().origin
+	camera.global_translation = interp_translation + head.translation
 
 
-remote func set_network_transform(new_translation: Vector3, new_rotation: Vector3):
-	translation = new_translation
-	rotation = new_rotation
+remote func set_network_transform(the_new_translation: Vector3, the_new_rotation: Vector3):
```

Comment:
Can these be renamed to just translation/rotation? Or does that conflict with the class members